### PR TITLE
Implement menu_create "ml" param to allow menu title/item to be looked up as multilingual key

### LINKED ISSUE #590: Feature Request: ML Keys for static menus & items titles
# Help us help you
  - [x] I have checked that my issue [doesn't exist yet](https://github.com/alliedmodders/amxmodx/issues).
  - [x] I have tried my absolute best to reduce the problem-space and have provided the absolute smallest test-case possible.
  - [x] I can always reproduce the issue with the provided description below.

# Environment
  * Operating System version: _Any version_
  * Game/AppID (with version if applicable): cstrike & another
  * Current AMX Mod X version: _below_
```
amxx version
AMX Mod X 1.9.0.5216 (http://www.amxmodx.org)
Authors:
        David "BAILOPAN" Anderson, Pavol "PM OnoTo" Marko
        Felix "SniperBeamer" Geyer, Jonny "Got His Gun" Bergstrom
        Lukasz "SidLuke" Wlasinski, Christian "Basic-Master" Hammacher
        Borja "faluco" Ferrer, Scott "DS" Ehlert
Compiled: Sep 10 2018 16:28:34
Built from: https://github.com/alliedmodders/amxmodx/commit/91f36f23
Build ID: 5216:91f36f23
Core mode: JIT+ASM32
```
  * Current Metamod version: below
```
meta version
Metamod-r v1.3.0.128, API (5:13)
Metamod-r build: 15:47:38 Aug 24 2018
Metamod-r from: https://github.com/theAsmodai/metamod-r/commit/0cf2f70
```
  - [x] I have updated AMX Mod X to the [latest version](https://www.amxmodx.org/downloads.php) and it still happens.
  - [x] I have updated AMX Mod X to the [latest snapshot](https://www.amxmodx.org/snapshots.php) and it still happens.
  - [x] I have updated Metamod to the [latest version](https://www.amxmodx.org/downloads.php) and it still happens.

# Description
I have a suggestion for creating static menus. 
We may use the 3rd argument of an native [menu_create](https://www.amxmodx.org/api/newmenus/menu_create). And when we use `.ml = true` - menu should use title names as ML Keys. Simple code example below.

# Problematic Code (or Steps to Reproduce)
  ```PAWN
  #include <amxmodx>

new g_hMenu;
new g_hMenuCB;

public plugin_init()
{
    register_plugin("Static menu ML tests", "0.0.1", "wopox1337");
    register_dictionary("static_menu_test.txt");

    Build_StaticMenu();

    register_clcmd("radio1", "CMD_MenuTest");
    // register_clcmd("say /MenuTest", "CMD_MenuTest");
}

Build_StaticMenu()
{
    g_hMenu = menu_create("MENU_NAME", "MenuHandler_1", .ml = true); // ml parameter not used now.
    g_hMenuCB = menu_makecallback("MenuCallback");

/*
 * This is way how i want to create items & menus. And ML at this moment don't works.
 */
    menu_additem(g_hMenu, "ITEM_0", .callback = g_hMenuCB);
    menu_additem(g_hMenu, "ITEM_1", .callback = g_hMenuCB);
    menu_additem(g_hMenu, "ITEM_2", .callback = g_hMenuCB);
    menu_additem(g_hMenu, "ITEM_3", .callback = g_hMenuCB);
    menu_additem(g_hMenu, "ITEM_4", .callback = g_hMenuCB);
}

public MenuCallback(id, menu, item)
{
    new name[32], _dummy[1];
    menu_item_getinfo(menu, item, _dummy[0], _dummy, charsmax(_dummy), name, charsmax(name), _dummy[0]);

/** This is wrong way to dinamicly set item name cuz name
 * will be re-setted each time & ML Key willn't found.
 */
//    menu_item_setname(menu, item, fmt("%l", name));
}

public MenuHandler_1(id, menu, item)
{
    // server_print("-> MenuHandler_1(%i, %i, %i)", id, menu, item);
}

public CMD_MenuTest(id)
{
    menu_display(id, g_hMenu);
    return PLUGIN_HANDLED;
}
  ```

# Logs
_Not needed now._

### PR DESCRIPTION
**Pull request by Arkshine**
Fixes #590.

When the menu displays, the menu title and items will be looked up as multilingual keys.
If the key doesn't exist, the menu item remains unchanged.

The `ml` param is around since the beginning I believe. It should not be needed to check whether the param exists or not.

Random test plugin:

```SourcePawn
#include <amxmodx>

new StaticMenuHandle;

public plugin_init()
{
    buildStaticMenu();

    register_dictionary("menutest.txt");
    register_clcmd("menu", "@OnCommandDisplayMenu");
}

public plugin_end()
{
    StaticMenuHandle && menu_destroy(StaticMenuHandle);
}

@OnCommandDisplayMenu(const player)
{
    menu_display(player, StaticMenuHandle);

    return PLUGIN_HANDLED;
}

buildStaticMenu()
{
    new const menu = menu_create("MENU_NAME", "@OnStaticMenu", .ml = true);

    menu_additem(menu, "ITEM_1");
    menu_additem(menu, "ITEM_2");
    menu_additem(menu, "ITEM_3");
    menu_additem(menu, "ITEM_4");

    StaticMenuHandle = menu;
}

@OnStaticMenu(const player, const menu, const item)
{

}
```
```
[en]
MENU_NAME = My Awesome Menu
ITEM_1    = My Item 1
ITEM_2    = My Item 2
ITEM_3    = My Item 3
ITEM_4    = My Item 4
```

Result:
image

If it happens keys don't exist:
image

### CODE REVIEW

File: amxmodx/newmenus.cpp [hunk 1]
```diff
@@ -620,38 +620,45 @@ const char *Menu::GetTextString(int player, page_t page, int &keys)
...
-	if (func == -1)
+	int length;
+	const auto title    = get_amxstring(amx, params[arg_title], 0, length);
+	const auto handler  = get_amxstring(amx, params[arg_handler], 1, length);
+	const auto callback = registerSPForwardByName(amx, handler, FP_CELL, FP_CELL, FP_CELL, FP_DONE);
+
+	if (callback == -1)
 	{
-		LogError(amx, AMX_ERR_NOTFOUND, "Invalid function \"%s\"", handler);
+		LogError(amx, AMX_ERR_NOTFOUND, R"(Invalid function %s)", handler);
```
Comment (WPMGPRoSToTeMa) [inline amxmodx/newmenus.cpp]: Inner quotes are missing.

Comment (Arkshine) [reply amxmodx/newmenus.cpp]: It's defining a literal string with `R` to remove the need to escape.

Comment (WPMGPRoSToTeMa) [reply amxmodx/newmenus.cpp]: @Arkshine I mean `%s`.

Comment (Arkshine) [reply amxmodx/newmenus.cpp]: Oh, right.

**Pull request merged**